components/textarea/counter.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useMemo } from 'react'
+import useTheme from '../styles/use-theme'
+
+interface CounterProps {
+  count: number
+  maxLength?: number
+}
+
+const defaultProps = { maxLength: undefined }
+
+const Counter: React.FC<React.PropsWithChildren<CounterProps>> = ({ count, maxLength }) => {
+  const theme = useTheme()
+  const hasLimit = useMemo(() => Number.isInteger(maxLength) && (maxLength as number) > 0, [
```

Comment:
has a bug: try to set maxLength to 1000000000000000000000

Reply:
done